src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    internal ref partial struct CborReader
```

Comment:
`CborReader` should be a class.  We can also have a `CborValueReader`, but the friendlier version should get the better name.

Reply:
I'm aware that this pattern exists in Asn1, however I chose to keep simple while features are being added.

Reply:
Sure, then class only, please.

Reply:
But that would require replacing span with memory, so more refactoring effort going back and forth. Should I just rename to `CborValueReader` for now?

Reply:
> Should I just rename to CborValueReader for now?

Oh, I suppose I can accept that compromise :smile:.

Reply:
Done.